Applications/TransferLearning/Draw_Classification/jni/main.cpp

```diff
@@ -318,30 +290,153 @@ int main(int argc, char *argv[]) {
-  //     NN.finalize();
-  //     return -1;
-  //   }
-  // }
+/**
+ * @brief Test the model with the given config file path
+ * @param[in] data_path Path of the test data
+ * @param[in] tflite_data TFLite loaded configuration
+ * @param[in] config Model config file path
+ */
+int testModel(std::string data_path, const TFLiteData &tflite_data,
+              const char *model) {
+#if defined(__TIZEN__)
+  int status = ML_ERROR_NONE;
+  ml_pipeline_h pipe;
+  ml_pipeline_src_h src;
+  ml_tensors_info_h in_info;
+  ml_tensors_data_h in_data;
+  void *raw_data;
+  size_t data_size;
+  ml_tensor_dimension in_dim = {1, 1, 1, INPUT_SIZE};
+
+  char pipeline[1024];
+  snprintf(
+    pipeline, sizeof(pipeline),
+    "appsrc name=srcx | "
+    "other/"
+    "tensor,dimension=(string)1:1:1:%d,type=(string)float,framerate=(fraction)"
+    "0/1 | tensor_filter framework=nntrainer model=%s | tensor_sink",
+    INPUT_SIZE, model);
+
+  status = ml_pipeline_construct(pipeline, NULL, NULL, &pipe);
+  if (status != ML_ERROR_NONE)
+    goto fail_exit;
+
+  status = ml_pipeline_src_get_handle(pipe, "srcx", &src);
+  if (status != ML_ERROR_NONE)
+    goto fail_pipe_destroy;
+
+  status = ml_pipeline_start(pipe);
+  if (status != ML_ERROR_NONE)
+    goto fail_src_release;
+
+  ml_tensors_info_create(&in_info);
+  ml_tensors_info_set_count(in_info, 1);
+  ml_tensors_info_set_tensor_type(in_info, 0, ML_TENSOR_TYPE_FLOAT32);
+  ml_tensors_info_set_tensor_dimension(in_info, 0, in_dim);
+
+  for (int i = 0; i < TOTAL_TEST_SIZE; i++) {
+    std::string path = data_path;
+    path += "testset";
+    printf("\n[%s]\n", path.c_str());
+    std::string img = path + "/";
+    img += "test" + std::to_string(i + 1) + ".bmp";
+    printf("%s\n", img.c_str());
+
+    std::vector<float> featureVector;
+    featureVector.resize(INPUT_SIZE);
+    getInputFeature(tflite_data, img, featureVector);
+
+    status = ml_tensors_data_create(in_info, &in_data);
+    if (status != ML_ERROR_NONE)
+      goto fail_info_release;
+
+    status = ml_tensors_data_get_tensor_data(in_data, 0, &raw_data, &data_size);
+    if (status != ML_ERROR_NONE || data_size != INPUT_SIZE) {
+      ml_tensors_data_destroy(&in_data);
+      goto fail_info_release;
+    }
+
+    for (size_t ds = 0; ds < data_size; ds++) {
+      ((float *)raw_data)[i] = featureVector[i];
+    }
+
+    status = ml_pipeline_src_input_data(src, in_data,
+                                        ML_PIPELINE_BUF_POLICY_AUTO_FREE);
+    if (status != ML_ERROR_NONE || data_size != INPUT_SIZE) {
+      ml_tensors_data_destroy(&in_data);
+      goto fail_info_release;
+    }
+  }
+
+fail_info_release:
+  ml_tensors_info_destroy(in_info);
+
+  status = ml_pipeline_stop(pipe);
+
+fail_src_release:
+  status = ml_pipeline_src_release_handle(src);
+
+fail_pipe_destroy:
+  status = ml_pipeline_destroy(pipe);
+
+fail_exit:
+  return status;
+#else
+  std::cerr << "Testing of model only with TIZEN" << std::endl;
+  return ML_ERROR_NONE;
```

Comment:
return error?

Reply:
I dont want the application to fail in non-tizen environment.
Rather print the error and just finish without inference.

Reply:
Alright